### ISSUE TITLE
ReferenceUsedNamesOnly - some annotations not detected, possibly when referenced native types exist

### ISSUE THREAD

**Issue by Majkl578** [slevomat/coding-standard#273]
The `@param` annotation in this snippet is not detected:
```php
<?php

declare(strict_types=1);

namespace Doctrine\ORM\Repository;

use Doctrine\Common\Persistence\ObjectRepository;
use Doctrine\ORM\EntityManagerInterface;

/**
 * Interface for entity repository factory.
 */
interface RepositoryFactory
{
    /**
     * Gets the repository for an entity class.
     *
     * @param \Doctrine\ORM\EntityManagerInterface $entityManager The EntityManager instance.
     * @param string                               $entityName    The name of the entity.
     *
     * @return ObjectRepository
     */
    public function getRepository(EntityManagerInterface $entityManager, $entityName);
}
```
When native type hint is removed, detection works.

And another case:
```php
<?php

declare(strict_types=1);

namespace Doctrine\ORM\Query\AST\Functions;

use Doctrine\ORM\Query\AST\Node;
use Doctrine\ORM\Query\Parser;
use Doctrine\ORM\Query\SqlWalker;

/**
 * Abstract Function Node.
 */
abstract class FunctionNode extends Node
{
    /**
     * @var string
     */
    public $name;

    /**
     * @param string $name
     */
    public function __construct($name)
    {
        $this->name = $name;
    }

    /**
     *
     * @return string
     */
    abstract public function getSql(SqlWalker $sqlWalker);

    /**
     * @param \Doctrine\ORM\Query\SqlWalker $sqlWalker
     *
     * @return string
     */
    public function dispatch($sqlWalker)
    {
        return $sqlWalker->walkFunction($this);
    }

    /**
     *
     * @return void
     */
    abstract public function parse(Parser $parser);
}
```
When I remove method `getSql()` (or the type hint in its parameter), detections works.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/4c16a297703fc1ada21098166e32aacc7264923b
